src/web-experiments-types.ts

```diff
@@ -0,0 +1,38 @@
+export interface WebExperimentTransform {
+    attributes:
+        | {
+              name: string
+              value: string
+          }[]
+        | null
+    selector?: string
+    text?: string
+    html?: string
+    imgUrl?: string
+    className?: string
+}
+
+export type WebExperimentUrlMatchType = 'regex' | 'not_regex' | 'exact' | 'is_not' | 'icontains' | 'not_icontains'
+
+export interface WebExperimentVariant {
+    conditions: {
+        url?: string
+        urlMatchType?: WebExperimentUrlMatchType
+        utm: {
+            utm_source?: string
+            utm_medium?: string
+            utm_campaign?: string
+            utm_term?: string
+        }
+    } | null
```

Comment:
similar comment about making this payload undefined vs null as I had above; why extra an extra null state that needs more defensive coding?